api/v1/installation_types.go

```diff
@@ -95,6 +95,16 @@ type InstallationSpec struct {
 	// +optional
 	TyphaAffinity *TyphaAffinity `json:"typhaAffinity,omitempty"`
 
+	// TyphaNodeSelector is used to select the nodes on which to run Typha pods.
+	// +optional
+	TyphaNodeSelector map[string]string `json:"typhaNodeSelector,omitempty"`
```

Comment:
"do not use" is probably too strong - there are legitimate reasons to use this. But a warning of the form "If used, you must ensure there is sufficient room for Typha to be scheduled or else your cluster may stop functioning" is probably appropriate.

Reply:
In theory we could build in a release valve - if the number of nodes selected by this is not large enough, we error and ignore the selector. 

I don't think it's quite worth doing, but if we end up getting a lot of issues where people break their clusters due to misconfiguring this field, we can look into it.

Reply:
Here is the warning we have with the TyphaAffinity.RequiredDuringScheduling...
```
    // WARNING: Please note that if the affinity requirements specified by this field are not met at
    // scheduling time, the pod will NOT be scheduled onto the node.
    // There is no fallback to another affinity rules with this setting.
    // This may cause networking disruption or even catastrophic failure!
    // PreferredDuringSchedulingIgnoredDuringExecution should be used for affinity
    // unless there is a specific well understood reason to use RequiredDuringSchedulingIgnoredDuringExecution and
    // you can guarantee that the RequiredDuringSchedulingIgnoredDuringExecution will always have sufficient nodes to satisfy the requirement.  
```
I think we should have something similar here. I'm a bit opinionated here but if we don't want to include something that ominous for this one then I think we should reduce the warning on the other to the same level as what we use here.